pkg/auth/kubeconfig.go

```diff
@@ -230,7 +268,90 @@ func generateX509KubeConfig(ctx context.Context, cli kubernetes.Interface, cfg *
 	return genKubeConfig(cfg, &clientcmdapi.AuthInfo{
 		ClientKeyData:         keyBytes,
 		ClientCertificateData: csr.Status.Certificate,
-	}, nil), nil
+	}, nil)
+}
+
+func generateX509KubeConfigBeta(ctx context.Context, cli kubernetes.Interface, cfg *clientcmdapi.Config, writer io.Writer, opts *KubeConfigGenerateX509Options) (*clientcmdapi.Config, error) {
```

Comment:
I think generateX509KubeConfigBeta has lots of redundant code with generateX509KubeConfig.